parametric/apps/golang/otel_methods.go

```
@@ -0,0 +1,254 @@
+	return &OtelFlushSpansReturn{Success: success}, nil
+}
+
+func (s *apmClientServer) OtelFlushTraceStats(context.Context, *OtelFlushTraceStatsArgs) (*OtelFlushTraceStatsReturn, error) {
+	s.otelSpans = make(map[uint64]ot_api.Span)
+	return &OtelFlushTraceStatsReturn{}, nil
+}
+
+func (s *apmClientServer) OtelIsRecording(ctx context.Context, args *OtelIsRecordingArgs) (*OtelIsRecordingReturn, error) {
+	span, ok := s.otelSpans[args.SpanId]
+	if !ok {
+		fmt.Printf("IsRecording call failed, span with id=%s not found", args.SpanId)
+	}
+	return &OtelIsRecordingReturn{IsRecording: span.IsRecording()}, nil
+}
+
+func (s *apmClientServer) OtelSpanContext(ctx context.Context, args *OtelSpanContextArgs) (*OtelSpanContextReturn, error) {
+	span, ok := s.otelSpans[(args.SpanId)]
+	if !ok {
+		fmt.Printf("SpanContext call failed, span with id=%s not found", args.SpanId)
+	}
+	sctx := span.SpanContext()
+	return &OtelSpanContextReturn{
+		SpanId:     sctx.SpanID().String(),
+		TraceId:    sctx.TraceID().String(),
+		TraceFlags: sctx.TraceFlags().String(),
+		TraceState: sctx.TraceState().String(),
+		Remote:     sctx.IsRemote(),
+	}, nil
+}
+
+func (s *apmClientServer) OtelSetStatus(ctx context.Context, args *OtelSetStatusArgs) (*OtelSetStatusReturn, error) {
+	span, ok := s.otelSpans[args.SpanId]
+	if !ok {
+		fmt.Sprintf("SetStatus call failed, span with id=%d not found", args.SpanId)
+	}
+	if args.Code == "UNSET" {
+		span.SetStatus(codes.Unset, args.Description)
+	} else if args.Code == "ERROR" {
+		span.SetStatus(codes.Error, args.Description)
+	} else if args.Code == "OK" {
+		span.SetStatus(codes.Ok, args.Description)
+	} else {
+		fmt.Sprintf("Invalid code")
+	}
+	return &OtelSetStatusReturn{}, nil
+}
+
+func hex2int(hexStr string) uint64 {
+	// remove 0x suffix if found in the input string
+	cleaned := strings.Replace(hexStr, "0x", "", -1)
+
+	// base 16 for hexadecimal
+	result, _ := strconv.ParseUint(cleaned, 16, 64)
```

Should probably panic if an error occurs here, instead of ignoring it.